src/fheroes2/audio/audio_manager.cpp

```diff
@@ -568,18 +605,38 @@ namespace
                 continue;
             }
 
-            std::vector<ChannelAudioLoopEffectInfo> & existingEffects = foundSoundTypeIter->second;
+            std::vector<ChannelAudioLoopEffectInfo> & effectsToReplace = foundSoundTypeIter->second;
+
+            // Search for an existing sound which has the exact volume and angle.
+            for ( auto soundToAddIter = effectsToAdd.begin(); soundToAddIter != effectsToAdd.end(); ) {
+                auto exactSoundEffect = std::find( effectsToReplace.begin(), effectsToReplace.end(), *soundToAddIter );
+                if ( exactSoundEffect != effectsToReplace.end() ) {
+                    currentAudioLoopEffects[soundType].emplace_back( *exactSoundEffect );
+                    effectsToReplace.erase( exactSoundEffect );
+
+                    soundToAddIter = effectsToAdd.erase( soundToAddIter );
+                    continue;
+                }
+
+                ++soundToAddIter;
+            }
+
+            size_t effectsToReplaceCount = std::min( effectsToAdd.size(), effectsToReplace.size() );
+
+            while ( effectsToReplaceCount > 0 ) {
+                --effectsToReplaceCount;
 
-            size_t elementSize = std::min( effects.size(), existingEffects.size() );
-            while ( elementSize > 0 ) {
-                --elementSize;
+                // Find the most closest angles to those which are going to be added.
```

Comment:
Just a small correction.
```suggestion
                // Find the closest angles to those which are going to be added.
```